packages/push-notifications/src/PushNotificationsModule.ts

```diff
@@ -0,0 +1,53 @@
+import type { DeviceInfo } from './services'
+
+import { ConnectionService } from '@aries-framework/core'
+import { Dispatcher } from '@aries-framework/core/build/agent/Dispatcher'
+import { MessageSender } from '@aries-framework/core/build/agent/MessageSender'
+import { createOutboundMessage } from '@aries-framework/core/build/agent/helpers'
+import { Lifecycle, scoped } from 'tsyringe'
+
+import { PushNotificationsDeviceInfoHandler } from './handlers/PushNotificationsDeviceInfoHandler'
+import { PushNotificationsService } from './services'
+
+/**
+ * Module that exposes push notification get and set functionality
+ */
+@scoped(Lifecycle.ContainerScoped)
+export class PushNotificationsModule {
+  public constructor(
+    private pushNotificationService: PushNotificationsService,
+    private connectionService: ConnectionService,
+    private messageSender: MessageSender,
+    dispatcher: Dispatcher
+  ) {
+    this.registerHandlers(dispatcher)
+  }
+
+  /**
+   * Sends the native device info (token and vendor) to another agent via a `connectionId`
+   */
+  public async sendNativeDeviceInfo(connectionId: string, deviceInfo: DeviceInfo) {
+    const connection = await this.connectionService.getById(connectionId)
+    const message = this.pushNotificationService.createSetNativeDeviceInfo(deviceInfo)
+
+    connection.assertReady()
```

Comment:
I'd swap these around. The service doesn't store anything so it doesn't matter in this case, but if the connection is not ready we don't have to call the service